src/main/java/com/iota/iri/service/API.java

```diff
@@ -491,26 +487,41 @@ public static void incEllapsedTime_getTxToApprove(long ellapsedTime) {
         ellapsedTime_getTxToApprove += ellapsedTime;
     }
 
-    public synchronized Hash[] getTransactionToApproveStatement(final int depth, final String reference, final int numWalks) throws Exception {
-        int tipsToApprove = 2;
-        Hash[] tips = new Hash[tipsToApprove];
-        final SecureRandom random = new SecureRandom();
-        final int randomWalkCount = numWalks > maxRandomWalks || numWalks < 1 ? maxRandomWalks:numWalks;
+    public synchronized Hash[] getTransactionToApproveStatement(final int depth, final String reference, final List<String> referenceTips, final int tipsToApprove, final int numWalks) throws Exception {
+        final Hash[] tips;
+        final SecureRandom random;
+        final int randomWalkCount;
+        final List<Hash> referenceTipsHashes;
+        final Snapshot referenceSnapshot;
+
+        tips = new Hash[tipsToApprove];
```

Comment:
requesting `count`: 1 results in an error - as the returned `tips` size is 1, but the dto is sent `tips[1]`